### PR TITLE
Adding ingest_if_empty() method

### PR DESCRIPTION
**Pull request by bendk**
This handles the component side of
https://bugzilla.mozilla.org/show_bug.cgi?id=1863661

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due diligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by bendk** [comment]
The idea here is to run ingest ahead of the regular 24-hour schedule when we do a schema upgrade that throws away the data.  My thought is that we can:

- Schedule `ingest_if_empty()` soon after startup.  Most of the time it will be a no-op, but it will run in the schema upgrade case and also on first run.
- Keep the current 24-hour schedule for `ingest()`, tweak it so that it waits 24-hours before the first run.

The bugzilla has more details.

File: components/suggest/src/suggest.udl [hunk 1]
```diff
@@ -137,6 +137,17 @@ interface SuggestStore {
...
+    // Run ingest, but only if the database is empty.
+    //
+    // This is indented to be used alongside ingest() to handle periodic updates.  Consumers can
+    // schedule ingest_if_empty() to run immediately on startup and for ingest() to run on a long
+    // periodic schedule (maybe once a day).
+    //
+    // This allows ingestion to normally be run at a slow, periodic rate, but when we do a schema
+    // upgrade and throw away the database then we will ingest the data immediately.
+    [Throws=SuggestApiError]
+    void ingest_if_empty(SuggestIngestionConstraints constraints);
```
[the remark this answers is not in the document]
Comment (linabutler) [reply components/suggest/src/suggest.udl:149]: If you're open to a suggestion, you can add `force_if_empty` to `SuggestIngestionConstraints`! It's there to let clients customize the ingestion behavior, and this seems like a great fit for that 😊

**Comment by bendk** [comment]
That's a great suggestion, just updated the PR.  I named the field `empty_only`, since normally the ingestion would run and that flag forces it not to.

**Review by ncloudioj — APPROVED** [review]
LGTM, just a few nits.

**Pull request merged**